engine/immediate_task_concurrent_processor.go

```diff
@@ -27,14 +27,14 @@ type immediateTaskConcurrentProcessor struct {
 	rootCtx           context.Context
 	cfg               config.Config
 	taskToProcessChan chan data_models.ImmediateTask
-	// for quickly checking if the shardId is being processed
-	currentShards map[int32]bool
-	// shardId to the channel
+	// shardId: channel
 	taskToCommitChans map[int32]chan<- data_models.ImmediateTask
-	taskNotifier      TaskNotifier
-	processStore      persistence.ProcessStore
-	visibilityStore   persistence.VisibilityStore
-	logger            log.Logger
+	// shardId: WaitForProcessCompletionChannelsPerShard
+	waitForProcessCompletionChannelsPerShardMap map[int32]WaitForProcessCompletionChannelsPerShard
```

Comment:
nit: WaitForProcessCompletionChannelsPerShard -> WaitForProcessCompletionChannels. the type doesn't need to know it's being used as "per shard". the var name is a better place.